pkgs/by-name/ri/rimsort/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  copyDesktopItems,
+  makeDesktopItem,
+  imagemagick,
+  stdenv,
+  todds,
+  steamPackages,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "rimsort";
+  version = "0-unstable-2024-04-20";
+
+  src = fetchFromGitHub {
+    owner = "RimSort";
+    repo = "RimSort";
+    rev = "794a20f99a499edbd98a0c15ea425ad86c72ca1b";
+    hash = "sha256-5ldheNcItNfXRysAGoCi8djLfCfmwLDfW8a3mFjzUQs=";
+  };
+
+  patches = [
+    ./make-nix-compatible.patch
```

Comment:
Can you try to upstream the generic fixes in there?

Reply:
@SuperSandro2000 this was my big question, don't know what fixes would be accepted or on what timeline etc.

i can try, i dont mind this pr sitting here people can still use it anyways, i just wasn't sure if it'd be preferable to get it working with ugly patches, get it merged, eventually get stuff upstreamed, then remove patches when the package gets updated, or to try to upstream 1st before ever merging

in either case i am going to try to upstream what i can

Reply:
Any progress on that?

Reply:
@nyabinary no. the rimsort guy didn't reply to my messages about this specifically and seemingly deleted another nix-related thread. 

i redid my patches in a way that is much cleaner but it also turned out to be significantly more lines changed to make it pretty

i'm just going to close this pr for now because i realize im not interested in maintaining rimsort as it is now. i'll leave my branch alive on my fork in case either someone else wants to pick it up or if eventually the generic changes can be upstreamed

Reply:
Hello. I am the guy you speak of. I am not sure what you are referring to that i am supposed to have deleted or not responded to.

I am open to supporting NixOS upstream, but am only willing to support code that we work on upstream. I am always open to discussing further, however limited my direct involvement in implementing may be.

I am not really trying to provide support for these patches specifically, and i think that some of what i have looked at warrants discussion. Some of those comments and observations i initially had probably got lost on the discord.